test/src/databases/all/functions.spec.ts

```diff
@@ -0,0 +1,1071 @@
+      expect(result.data.path(1, 'a').value).toBe(
+        result.data.path(0, 's').value
+      );
+    });
+  });
+
+  describe('first_value', () => {
+    it(`works in nest - ${databaseName}`, async () => {
+      const result = await expressionModel
+        .loadQuery(
+          `
+          query: aircraft -> {
+            group_by: state
+            where: state != null
+            nest: by_county is {
+              limit: 2
+              group_by: county
+              aggregate: aircraft_count
+              calculate: first_count is first_value(count())
+            }
+          }`
+        )
+        .run();
+      expect(result.data.path(0, 'by_county', 1, 'first_count').value).toBe(
+        result.data.path(0, 'by_county', 0, 'aircraft_count').value
+      );
+      expect(result.data.path(1, 'by_county', 1, 'first_count').value).toBe(
+        result.data.path(1, 'by_county', 0, 'aircraft_count').value
+      );
+    });
+    it(`works outside nest - ${databaseName}`, async () => {
+      const result = await expressionModel
+        .loadQuery(
+          `
+          query: state_facts -> {
+            group_by: state, births
+            order_by: births desc
+            calculate: most_births is first_value(births)
+          }`
+        )
+        .run();
+      const firstBirths = result.data.path(0, 'births').value;
+      expect(result.data.path(0, 'most_births').value).toBe(firstBirths);
+      expect(result.data.path(1, 'most_births').value).toBe(firstBirths);
+    });
+    it(`works with an aggregate which is not in the query - ${databaseName}`, async () => {
+      const result = await expressionModel
+        .loadQuery(
+          `
+          query: airports { measure: airport_count is count() } -> {
+            group_by: state
+            where: state != null
+            calculate: prev_airport_count is lag(airport_count)
+          }`
+        )
+        .run();
+      expect(result.data.path(0, 'prev_airport_count').value).toBe(null);
+      expect(result.data.path(1, 'prev_airport_count').value).toBe(608);
+      expect(result.data.path(2, 'prev_airport_count').value).toBe(260);
+    });
+    it(`works with a localized aggregate - ${databaseName}`, async () => {
+      const result = await expressionModel
+        .loadQuery(
+          `
+          query: aircraft -> {
+            group_by: aircraft_models.seats,
+            calculate: prev_sum_of_seats is lag(aircraft_models.seats.sum())
+          }`
+        )
+        .run();
+      expect(result.data.path(0, 'prev_sum_of_seats').value).toBe(null);
+      expect(result.data.path(1, 'prev_sum_of_seats').value).toBe(0);
+      expect(result.data.path(2, 'prev_sum_of_seats').value).toBe(230);
+    });
+  });
+
+  describe('trunc', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['trunc(1.9)', 1],
+        // TODO Remove when we upgrade to DuckDB 0.8.X -- DuckDB has some bugs with rounding
```

Comment:
A bunch of tests fail for DuckDB WASM right now, so I've skipped them. I think they should all be fixed once we can upgrade to 0.8.